src/pngimage.cpp

```diff
@@ -533,16 +552,17 @@ namespace Exiv2 {
         // Write PNG Signature.
         if (outIo.write(pngSignature, 8) != 8) throw Error(kerImageWriteFailed);
 
-        DataBuf cheaderBuf(8);       // Chunk header : 4 bytes (data size) + 4 bytes (chunk type).
+        DataBuf cheaderBuf(9);   // Chunk header : 4 bytes (data size) + 4 bytes (chunk type) + nul
```

Comment:
I do not understand why we need to do this change.

You are changing the size from 8 to 9 but then the last byte is never used.
In line 556 you are setting byte[8] to 0, but that is already done in line 562 where we set the whole byte array to 0. 

Could you explain why this is needed?

Reply:
I did this to use strcmp() instead of memcmp() to inspect different chunks.  You can revert that, if you prefer memcmp.()

Reply:
In my opinion, I think this change is not needed . We are passing `szChunk` to the  `strcmp` function and not `cheaderBuf`.  You can still copy 4 bytes from cheaderBuf to szChunk and szChunk can have the 5th byte set to NULL.